src/main/kotlin/Leader.kt

```diff
@@ -1,5 +1,9 @@
 package org.example
 
+
+//TODO maybe we should make all the constructors except the Followe as private
+// this way all nodes can only be initialized as Follower and only transition to a new
+// state thought the state machine
```

Comment:
Interesting. Not quite sure what you mean, we should make a `StateMachine` wrapper, or have the constructors be protected or whatever the equivalent is, so it's only available within the package?

Reply:
Having the constructors protected. The Node knows how to transition from one state to another so no need for a StateMachine wrapper.

Reply:
We leave only the Follower as public, as no other Node can be initiated in a different state.